minetest/minetest
# Improve default inventory+wield images of node drawtypes

This PR improves the default inventory and wield images of many node drawtypes (when `inventory_image` and `wield_image` are `nil`).

Attends to #9209 (partial fix).
Fixes #9256.
Fixes #9209.

## Drawtype changes

* `airlike`: Fallback image added (modder is still adviced to add custom inventory/wield image)
* `signlike`: New rotation, backface rendered, new wield image
* `torchlike`: New rotation, backface rendered, new wield image
* `plantlike`, `plantlike_rooted`: Use actual node appearance
* `firelike`: Backfaces rendered
* `flowingliquid`: flat node, new wield image

## Problems fixed

* `airlike` was invisible
* Backfaces of `signlike`, `torchlike`, `plantlike`, `firelike` were not rendered

## Other changes

* Wallmounted nodes are rotated to a more appropriate position

## NOT fixed

* The base cube of `plantlike_rooted` is still not rendered

This is the only thing not fixed for #9209.

## How to test

Use `devtest` https://git.minetest.land/Wuzzy/devtest. This contains test nodes for all drawtypes, all with `inventory_image` and `wield_image` being set to `nil`, therefore the engine default will be used.

First, do the test *without* this PR to see the current stage so you know what will change.

Place the chest of everything (`chest_of_everything:chest`) and click through the items. Check out both wield and inventory images. Also check with `inventory_items_animations=true`.

Then, apply this PR and do the test again.
